src/request.ts

```diff
@@ -1330,45 +1325,53 @@ export type ProjectId = string | null | undefined;
 export interface RequestCallback {
   (
     a?: Error | null,
-    b?: AllocateIdsRequestResponse &
+    b?: google.datastore.v1.IAllocateIdsResponse &
       google.datastore.v1.ILookupResponse &
-      Entities
+      google.datastore.v1.ICommitResponse &
+      google.datastore.v1.IRollbackResponse & {
+        transaction?: string;
+      }
   ): void;
 }
 export interface RequestConfig {
   client: string;
-  gaxOpts?: number | CallOptions | KeyProto;
+  gaxOpts?: CallOptions;
   method: string;
   prepared?: boolean;
-  reqOpts?: Entity | RequestOptions;
-  gaxOptions?: never;
+  reqOpts?: RequestOptions;
 }
 export interface RequestOptions {
-  mutations?: [] | Array<{delete: KeyProto}> | Array<{}>;
+  mutations?: google.datastore.v1.IMutation[];
   keys?: Entity;
-  readOptions?: {readConsistency?: number; transaction?: string | number};
+  readOptions?: {readConsistency?: number; transaction?: string};
+  partitionId?: google.datastore.v1.IPartitionId | null;
   transactionOptions?: {
     readOnly?: {};
     readWrite?: {previousTransaction?: string};
-  };
-  transaction?: string | number;
+  } | null;
+  transaction?: string | null;
   mode?: string;
-  projectId?: string;
-}
-export interface RunQueryStreamOptions {
-  gaxOptions?: CallOptions;
-  consistency?: 'strong' | 'eventual';
+  projectId?: ProjectId;
+  query?: QueryProto;
 }
-export interface SaveCallback {
-  (a?: Error | null, b?: Entity): void;
+export interface RunQueryStreamOptions extends RunQueryOptions {}
+export interface CommitCallback {
+  (err?: Error | null, resp?: google.datastore.v1.ICommitResponse): void;
 }
+export type CommitResponse = [google.datastore.v1.ICommitResponse];
+export interface SaveCallback extends CommitCallback {}
+export type SaveResponse = CommitResponse;
+export interface UpdateCallback extends CommitCallback {}
+export type UpdateResponse = CommitResponse;
+export interface UpsertCallback extends CommitCallback {}
+export type UpsertResponse = CommitResponse;
 
 /*! Developer Documentation
  *
  * All async methods (except for streams) will return a Promise in the event
  * that a callback is omitted.
  */
-promisifyAll(DatastoreRequest);
+promisifyAll(DatastoreRequest, {exclude: ['request_']});
```

Comment:
Do we need this change? I thought that it excluded methods with trailing underscores by default?

Reply:
You are right, forgot about it. 
Not needed!